libkbfs/favorites.go

```diff
@@ -107,6 +112,87 @@ func NewFavorites(config Config) *Favorites {
 	return newFavoritesWithChan(config, make(chan *favReq, 100))
 }
 
+func (f *Favorites) readCacheFromDisk(ctx context.Context) error {
+	// Read the encrypted cache from disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	// Send the data to the service to be decrypted
+	decryptedData, err := f.config.KeybaseService().DecryptFavorites(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	// Decode the data into the a map
+	decoder := gob.NewDecoder(bytes.NewBuffer(decryptedData))
+	var cache map[Favorite]bool
+	err = decoder.Decode(&cache)
+	if err != nil {
+		return err
+	}
+
+	f.cache = cache
+	return nil
+}
+
+func (f *Favorites) writeCacheToDisk(ctx context.Context) error {
+	// Encode the cache map into a byte buffer
+	buf := &bytes.Buffer{}
+	encoder := gob.NewEncoder(buf)
+	err := encoder.Encode(f.cache)
+	if err != nil {
+		return err
+	}
+
+	// Send the byte buffer to the service for encryption
+	data, err := f.config.KeybaseService().EncryptFavorites(ctx, buf.Bytes())
+	if err != nil {
+		return err
+	}
+
+	// Write the encrypted cache to disk
+	folder := f.config.StorageRoot()
+	subfolder := "kbfs.favorites.cache"
+	session, err := f.config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	user := string(session.UID)
+	path := filepath.Join(folder, subfolder, user)
+	// TODO: is this the right permissions
+	return ioutil.WriteFile(path, data, 0600)
```

Comment:
Besides using msgpack for the encoded data we send over to the service (as mentioned above), I think we probably want to also use it here for the contents of the file on disk (i.e., a struct that contains a `[]byte`).  That's because we should also add a version number to that struct also, in case we ever change the encryption scheme of the cache in the future.  And when reading the file, check the version number and panic if it's unexpected.